RFS/src/main/java/com/rfs/cms/ProcessManager.java

```diff
@@ -0,0 +1,51 @@
+package com.rfs.cms;
+
+import io.netty.util.concurrent.DefaultThreadFactory;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+/**
+ * This class takes an expiration time and guarantees that the process will NOT run past that expiration
+ * time unless the work is marked as complete before that expiration.  This class may, but does not need to,
+ * synchronize its clock with an external source of truth for better accuracy.
+ */
+public class ProcessManager {
```

Comment:
In the language of this code base this is a WorkManager - process isn't a noun, right?

Reply:
Renamed to LeaseExpireTrigger since that's what it's really doing